src/core/Akka.Streams.Tests/Dsl/FlowLogSpec.cs

```diff
@@ -26,145 +28,169 @@ public FlowLogSpec(ITestOutputHelper helper) : base("akka.loglevel = DEBUG", hel
         {
             var settings = ActorMaterializerSettings.Create(Sys).WithInputBuffer(2, 16);
             Materializer = ActorMaterializer.Create(Sys, settings);
+        }
+
+        private TestProbe _logProbe;
 
+        protected override void AtStartup()
+        {
+            base.AtStartup();
+            
             var p = CreateTestProbe();
-            Sys.EventStream.Subscribe(p.Ref, typeof(object));
-            LogProbe = p;
+            Sys.EventStream.Subscribe(p.Ref, typeof(LogEvent));
+            _logProbe = p;
         }
 
-        private TestProbe LogProbe { get; }
+        private async Task<string[]> LogMessages(int n, string tag)
+        {
+            return (await LogEvents<Debug>(n, tag)).Select(m => m.Message.ToString()).ToArray();
+        }
 
-        private string[] LogMessages(int n)
-            => LogProbe.ReceiveN(n).Cast<Debug>().Select(x => x.Message.ToString()).ToArray();
+        private async Task<T[]> LogEvents<T>(int n, string tag) where T : LogEvent
+        {
+            var count = 0;
+            var messages = new List<T>();
+            while (count < n)
+            {
+                var msg = (T) await _logProbe.FishForMessageAsync(m => m is T d && d.Message.ToString().StartsWith(tag));
+                count++;
+                messages.Add(msg);
+            }
+            return messages.ToArray();
+        }
```

Comment:
custom "ReceiveN" helper, instead of blindly pulling messages, limit the messages to the explicit type and tags that we know must be emitted